### PR TITLE
Engineer new features for convokit

### PR DESCRIPTION
**Pull request by chanteriam**
## Describe your changes
Added average sentence count and average word count to token aggregations.

## Checklist before requesting a review

- [x] The code runs successfully.

```
Here is some console output
```
- [x] I ran `make lint` and have made the bulk of changes that it has requested.

### CODE REVIEW

File: supreme_court_predictions/processing/token_aggregation.py [hunk 1]
```diff
@@ -40,6 +44,44 @@ def __init__(self, debug_mode=False):
         self.utterances = self.get_utterances()
         self.utterance_sides = self.append_side(self.utterances)
 
+    @staticmethod
+    def get_sentence_count(text):
+        """
+        Calculates the number of sentences and the number of words per sentence
```
Comment (michplunkett) [inline supreme_court_predictions/processing/token_aggregation.py:50]: 🔬  Add a docstring for the `text`.

File: supreme_court_predictions/processing/token_aggregation.py [hunk 2]
```diff
@@ -40,6 +44,44 @@ def __init__(self, debug_mode=False):
...
+        text_lst = text.split(SPLIT_PUNC)
+        text_lst = [
+            sen.strip() for sen in text_lst if len(sen) >= MIN_SENTENCE_LEN
+        ]
+
+        return len(text_lst)
+
+    def engineer_features(self, ut):
+        """
+        Engineers utterance count features for the utterances dataset.
```
Comment (michplunkett) [inline supreme_court_predictions/processing/token_aggregation.py:69]: 🔬 Add a docstring for the `ut`.

File: supreme_court_predictions/processing/token_aggregation.py [hunk 3]
```diff
@@ -193,21 +267,24 @@ def get_judge_case_tokens(self):
...
-        advocate_cts = (
-            advocate_cts.loc[:, ["case_id", "for_resp", "for_pet"]]
-            .groupby("case_id")
-            .agg("sum")
+        judge_tokens = self.get_case_tokens(
+            ut.loc[:, ["case_id", "tokens", "num_sentences", "num_words"]]
         )
 
-        return pd.merge(advocate_cts, judge_tokens, on="case_id")
+        # Add advocate counts to tokens
```
[the remark this answers is not in the document]
Comment (chanteriam) [reply supreme_court_predictions/processing/token_aggregation.py:274]: No, it was generally 1 advocate, 1 adversary per so it was useless

File: supreme_court_predictions/processing/token_aggregation.py (hunk 1 above)
Comment (chanteriam) [reply supreme_court_predictions/processing/token_aggregation.py:50]: u rights

File: supreme_court_predictions/processing/token_aggregation.py (hunk 2 above)
Comment (chanteriam) [reply supreme_court_predictions/processing/token_aggregation.py:69]: u right

File: supreme_court_predictions/processing/token_aggregation.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (michplunkett) [reply supreme_court_predictions/processing/token_aggregation.py:274]: Cool, I'd just remove them in this case. They're in the history if we need to access them again.

**Review by FedericoDM — APPROVED** [review]
:)

File: supreme_court_predictions/processing/token_aggregation.py [hunk 4]
```diff
@@ -40,6 +44,50 @@ def __init__(self, debug_mode=False):
         self.utterances = self.get_utterances()
         self.utterance_sides = self.append_side(self.utterances)
 
+    @staticmethod
+    def get_sentence_count(text):
+        """
+        Calculates the number of sentences and the number of words per sentence
```
Comment (michplunkett) [inline supreme_court_predictions/processing/token_aggregation.py:50]: You're missing the input param part of it tho. 😱

**Review by michplunkett — APPROVED** [review]

File: supreme_court_predictions/processing/token_aggregation.py (hunk 4 above)
Comment (chanteriam) [reply supreme_court_predictions/processing/token_aggregation.py:50]: i am DUM michael pls

**Pull request merged**